packages/react-badge/src/components/CounterBadge/CounterBadge.types.ts

```diff
@@ -1,71 +1,49 @@
 import type { BadgeProps, BadgeState } from '../Badge/index';
 
-/**
- * {@docCategory CounterBadge}
- */
-export interface CounterBadgeProps extends Omit<BadgeProps, 'appearance' | 'shape'> {
-  /**
-   * A Badge can be circular or rounded
-   * @defaultvalue circular
-   */
-  shape?: Extract<BadgeProps['shape'], 'rounded' | 'circular'>;
-
-  /**
-   * A Badge can be filled, ghost
-   * @defaultvalue filled
-   */
-  appearance?: Extract<BadgeProps['appearance'], 'filled' | 'ghost'>;
-
+export interface CounterBadgeCommons {
   /**
    * Max number to be displayed
-   * @defaultvalue 99
+   * @default 99
    */
-  overflowCount?: number;
+  overflowCount: number;
 
   /**
    * Value displayed by the Badge
-   * @defaultvalue 0
+   * @default 0
    */
-  count?: number;
+  count: number;
 
   /**
    * If the badge should be shown when count is 0
-   * @defaultvalue false
+   * @default false
    */
-  showZero?: boolean;
+  showZero: boolean;
 
   /**
    * If a dot badge should be displayed
-   * @defaultvalue false
+   * @default false
    */
-  dot?: boolean;
-}
+  dot: boolean;
 
-/**
- * {@docCategory CounterBadge}
- */
-export interface CounterBadgeState extends BadgeState {
   /**
-   * Max number to be displayed
-   * @defaultvalue 99
+   * A Badge can be circular or rounded
+   * @default circular
    */
-  overflowCount: number;
+  shape: Extract<BadgeProps['shape'], 'rounded' | 'circular'>;
 
   /**
-   * Value displayed by the Badge
-   * @defaultvalue 0
+   * A Badge can be filled, ghost
+   * @default filled
    */
-  count: number;
+  appearance: Extract<BadgeProps['appearance'], 'filled' | 'ghost'>;
+}
 
-  /**
-   * If the badge should be shown when count is 0
-   * @defaultvalue false
-   */
-  showZero: boolean;
+/**
+ * {@docCategory CounterBadge}
+ */
+export interface CounterBadgeProps extends Omit<BadgeProps, 'appearance' | 'shape'>, Partial<CounterBadgeCommons> {}
```

Comment:
I just don't see a need for extra types at this point, is there a reason it would be better to have separate types ? is it for API consistency ?

Reply:
The `CounterBadgeSlots` type is purely for consistency. It could be left out if desired.

The other half of my suggestion is to have `CounterBadgeCommons` extend `BadgeCommons` types extend each other, rather than having the Props/State types extend the base badge. That makes it so you only need to use `Omit` for `'appearance' | 'shape'` once, and it seems more logical to me, at least.